pkg/provisioners/natss/dispatcher/channel/reconcile.go

```diff
@@ -0,0 +1,162 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package channel
+
+import (
+	"context"
+
+	"github.com/knative/eventing/pkg/provisioners"
+	"github.com/knative/eventing/pkg/provisioners/natss/dispatcher/dispatcher"
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	eventingv1alpha1 "github.com/knative/eventing/pkg/apis/eventing/v1alpha1"
+	ccpcontroller "github.com/knative/eventing/pkg/provisioners/natss/controller/clusterchannelprovisioner"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type reconciler struct {
+	client   client.Client
+	recorder record.EventRecorder
+	logger   *zap.Logger
+
+	subscriptionsSupervisor *dispatcher.SubscriptionsSupervisor
+}
+
+// Verify the struct implements reconcile.Reconciler
+var _ reconcile.Reconciler = &reconciler{}
+
+func (r *reconciler) InjectClient(c client.Client) error {
+	r.client = c
+	return nil
+}
+
+func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	r.logger.Info("Reconcile: ", zap.Any("request", request))
+
+	ctx := context.TODO()
+	c := &eventingv1alpha1.Channel{}
+	err := r.client.Get(ctx, request.NamespacedName, c)
+
+	// The Channel may have been deleted since it was added to the workqueue.
+	if errors.IsNotFound(err) {
+		r.logger.Info("Could not find Channel", zap.Error(err))
+		// unsubscribe all active subscriptions for this channel
```

Comment:
I'm not sure we can rely on this to notice `Channel` deletion. I would expect to use a finalizer if you want to notice the channel is being deleted.

Reply:
Channel finalizer is used now to cleanup the NATSS active subscriptions.
Done